services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/InternalKafkaTopicCleanupService.java

```diff
@@ -0,0 +1,183 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.commandrouter.impl.kafka;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.eclipse.hono.client.kafka.HonoTopic;
+import org.eclipse.hono.client.kafka.KafkaAdminClientConfigProperties;
+import org.eclipse.hono.commandrouter.AdapterInstanceStatusService;
+import org.eclipse.hono.util.Lifecycle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.kafka.admin.KafkaAdminClient;
+
+/**
+ * A service to delete obsolete {@link HonoTopic.Type#COMMAND_INTERNAL} topics.
+ */
+public class InternalKafkaTopicCleanupService implements Lifecycle {
+
+    private static final Logger LOG = LoggerFactory.getLogger(InternalKafkaTopicCleanupService.class);
+
+    private static final long CHECK_INTERVAL_MILLIS = TimeUnit.MINUTES.toMillis(5);
```

Comment:
FMPOV we can set this to even higher value. In real time, I think not so often those internal topics become obsolete and should to be deleted too. We can avoid making the Kubernetes call and also Kafka call to list topics every 5 minutes by all the command router instances. IMHO later we can consider making this value configurable, so that it can help optimizing test/performance-test setups.

Reply:
I've now set the interval to 10min, so that obsolete topics will be deleted between 10 and 20 min after the corresponding adapter pods got shutdown. I think the K8s API requests won't matter much with these kinds of intervals.